backend/models/__init__.py

```diff
@@ -59,3 +61,59 @@ class TagOfItem(db.Model):  # type: ignore[name-defined]
         ),
         primary_key=True,
     )
+
+
+class ShoppingCart(db.Model):  # type: ignore[name-defined]
+    user_id = db.Column(
+        db.ForeignKey(User.uid, ondelete="CASCADE", onupdate="CASCADE"),
+        primary_key=True,
+    )
+    item_id = db.Column(
+        db.ForeignKey(Item.id, ondelete="CASCADE", onupdate="CASCADE"),
+        primary_key=True,
+    )
+    count = db.Column(db.Integer, nullable=False)
+
+    __table_args__ = (db.CheckConstraint(count > 0),)
+
+
+@enum.unique
+class OrderStatus(enum.Enum):
+    CANCEL = enum.auto()
+    CHECKING = enum.auto()
+    OK = enum.auto()
+
+
+@enum.unique
+class DeliveryStatus(enum.Enum):
+    DELIVERED = enum.auto()
+    DELIVERING = enum.auto()
+    PENDING = enum.auto()
+    SKIP = enum.auto()
+
+
+class Order(db.Model):  # type: ignore[name-defined]
+    order_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    user_id = db.Column(db.ForeignKey(User.uid, ondelete="CASCADE", onupdate="CASCADE"))
```

Comment:
基於訂單一定是由使用者所創立，故每筆訂單必定會有使用者，`user_id` 的 nullable 應為 NO。
```suggestion
    user_id = db.Column(db.ForeignKey(User.uid, ondelete="CASCADE", onupdate="CASCADE", nullable=False))
```

Reply:
已於 f26cbf814b2d648854095d8ec6240277709f242f 修正。
其中 nullable 是 `db.Column` 的參數而非 `db.ForeignKey` 的參數，因此進行了小調整。